### PR TITLE
Optionally filter out invalid markets

### PR DESCRIPTION
**Pull request by kongzii**
We have two functions that seem similar (I was thinking about merging them first), but serve a bit different purpose:

- `is_invalid` - this evaluates if the question should be resolved as invalid in the end; theoretically, we can have agent that doesn't care about it, e.g. this one https://github.com/gnosis/prediction-market-agent/issues/505
- `is_predictable_binary` - this evaluates if the question is predictable by the agent

To show the difference:

`Will X be alive at DD.MM.YYYY?` --> this is predictable, but it's invalid because it's violent
`What will be the price of GNO at DD.MM.YYYY?` -> this isn't predictable because our agents support binary markets only, but it's a valid question

### CODE REVIEW

File: tests/tools/test_is_invalid.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import pytest
+
+from prediction_market_agent_tooling.tools.is_invalid import is_invalid
+from tests.utils import RUN_PAID_TESTS
+
+
+@pytest.mark.skipif(not RUN_PAID_TESTS, reason="This test costs money to run.")
+@pytest.mark.parametrize(
+    "question, invalid",
```
Comment (kongzii) [inline tests/tools/test_is_invalid.py:9]: yay 
Screenshot by Dropbox Capture

File: prediction_market_agent_tooling/tools/is_invalid.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+)
+
+# I tried to make it return a JSON, but it didn't work well in combo with asking it to do chain of thought.
+# Rules are almost copy-pasted from https://cdn.kleros.link/ipfs/QmZM12kkguXFk2C94ykrKpambt4iUVKsVsxGxDEdLS68ws/omen-rules.pdf,
+# with some small prompting mods and I removed the point about "The outcome of the market must be known by its Resolution Date.", because that can not be verified before-hand.
+# and also point about "in which none of the answers are valid will resolve as invalid" and "in which multiple answers are valid will resolve as invalid.", because before hand we can not know if one of the outcomes happened or not.
+QUESTION_IS_INVALID_PROMPT = """Main signs about an invalid question (sometimes referred to as a "market"):
+- The market's question is about immoral violence, dead or assatination.
+- The violant event can be caused by a single conscious being.
+- The violant event is done ilegally.
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/tools/is_invalid.py:25]:
```suggestion
- The market's question is about immoral violence, death or assassination.
- The violent event can be caused by a single conscious being.
- The violent event is done illegally.
```

File: prediction_market_agent_tooling/tools/is_invalid.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+"""
+
+
+@persistent_inmemory_cache
+@tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_fixed(1))
+@observe()
+def is_invalid(
+    question: str,
+    engine: str = "gpt-4o",
+    temperature: float = LLM_SUPER_LOW_TEMPERATURE,
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/tools/is_invalid.py:62]: 🥶

**Review by evangriffiths — APPROVED** [review]
niiice!

File: prediction_market_agent_tooling/tools/is_invalid.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  - Valid: Will the US be engaged in a military conflict with a UN member state in 2021? (It’s unlikely for the US to declare war in order to win a bet on this market).
+  - Valid: Will Derek Chauvin go to jail for the murder of George Flyod? (It’s unlikely that the jurors would collude to make a wrong verdict in order to win this market).
+- Questions with relative dates will resolve as invalid. Dates must be stated in absolute terms, not relative depending on the current time.
+- Invalid: Who will be the president of the United States in 6 months? (“in 6 months depends on the current time”).
+- Questions about moral values and not facts will be resolved as invalid.
+- Invalid: “Is it ethical to eat meat?”.
+
+Follow a chain of thought to evaluate if the question is invalid:
+
+First, write the parts of the following question:
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/tools/is_invalid.py:44]: Just giving the LLM more tokens to reason about it. Completion looks like this:

Screenshot by Dropbox Capture

vs. if the question is more complicated

Screenshot by Dropbox Capture

this helped before in the `is_predictable_binary` function, I just kept it here.

**Review by evangriffiths — APPROVED** [review]

**Pull request merged**